driver/pace/driver/run.py

```diff
@@ -509,10 +545,39 @@ def step_all(self):
             self._step(timestep=self.config.timestep.total_seconds())
             time += self.config.timestep
             self.diagnostics.store(time=time, state=self.state)
+        self.timer.stop("total")
+        self._collect_performance()
+        self._write_out_performance()
 
     def _step(self, timestep: float):
-        self._step_dynamics(timestep=timestep)
-        self._step_physics(timestep=timestep)
+        with self.timestep_timer.clock("mainloop"):
+            self._step_dynamics(timestep=timestep)
+            self._step_physics(timestep=timestep)
+        self._collect_performance()
+        self.timestep_timer.reset()
+
+    def _collect_performance(self):
+        if self.config.performance_config.json_dump:
+            self.times_per_step.append(self.timestep_timer.times)
+            self.hits_per_step.append(self.timestep_timer.hits)
+
+    def _write_out_performance(self):
+        if self.config.performance_config.json_dump:
+            self.times_per_step.append(self.timer.times)
+            self.hits_per_step.append(self.timer.hits)
+            self.comm.Barrier()
+            while {} in self.hits_per_step:
+                self.hits_per_step.remove({})
+            collect_data_and_write_to_file(
+                len(self.hits_per_step) - 1,
+                self.config.stencil_config.backend,
+                self.config.git_hash,
+                self.comm,
+                self.hits_per_step,
+                self.times_per_step,
+                self.config.experiment_name,
+                self.config.dt_atmos,
+            )
```

Comment:
Can these concerns be put on the performance class instead of the driver class?

Reply:
Moved to `PerformanceConfig`.